packages/gatsby-theme-docs/src/modules/sso/hooks/use-authentication.ts

```diff
@@ -0,0 +1,46 @@
+import { useContext, useEffect, useState } from 'react';
+import { useAuth0 } from '@auth0/auth0-react';
+import useSWR from 'swr';
+import { fetcherWithToken } from '../../self-learning/hooks/hooks.utils';
+import { useAuthToken } from '../../self-learning/hooks/use-auth-token';
+import ConfigContext, {
+  EFeatureFlag,
+  isFeatureEnabled,
+} from '../../../components/config-context';
+
+const useAuthentication = () => {
+  const { isAuthenticated: isAuth0Authenticated, ...rest } = useAuth0();
+  const [isAuthenticated, setIsAuthenticated] = useState(false);
+  const { learnApiBaseUrl, selfLearningFeatures } = useContext(ConfigContext);
+  const { getAuthToken } = useAuthToken();
+
+  const shouldFetchData =
+    isFeatureEnabled(EFeatureFlag.CourseStatus, selfLearningFeatures) &&
+    isAuth0Authenticated;
+
+  const apiEndpoint = `/api/courses`;
```

Comment:
Was `/api/courses` chosen to be the one blocking sync call because it's relatively likely that it will be called anyways?  Or because it's not very heavy on the backend (which I don't know)? Or because the data is changing frequently anyways so the info would need to be updated anyways on every other call we do? 

It would be good to document the choice in the code, it's not very self-explanatory and the implication of revalidating it every three minutes for every user could be significant (the hook is called very often so you can assume this code running more or less nonstop).   Originally, it felt more natural to me to use the user profile as the default info we make available before anything else happens  but from a UI loading perspective the courses list may be a good choice, too.

Reply:
> Was `/api/courses` chosen to be the one blocking sync call because it's relatively likely that it will be called anyways? Or because it's not very heavy on the backend (which I don't know)? Or because the data is changing frequently anyways so the info would need to be updated anyways on every other call we do?

Various reasons. Firstly that endpoint doesn't get any parameters, so it can be called straight away without any additional logic. Another reason is that it's a sort of _staple_ endpoint for our use cases: you always need to know at least in which courses you're enrolled in before doing anything else.

> 
> It would be good to document the choice in the code, it's not very self-explanatory and the implication of revalidating it every three minutes for every user could be significant (the hook is called very often so you can assume this code running more or less nonstop). 

document it in the code: yes. for the rest, I'm not sure to understand. The call goes through swr which is ensuring no actual call is sent to the server as long as there's a valid cache. And for the additional flags. These are the standard set of flags we agreed (after discussing and testing it) to use in our application. In details:
```
revalidateOnReconnect: false; you suggested to leave it disabled as it's not luckly to happen as we don't have many mobile users
revalidateIfStale: false; it must be left false otherwise it get's triggered at any user navigation (due to "gatsby SPA" limitation that unmounts and remounts all the components at each navigation)
revalidateOnMount: true; this must left true for the initial load to happen (it doesn't trigger if a cache is present)
dedupingInterval: 3 minutes; we decided to add this arbitrary timeout as fallback. If not set or set too high, the only way we have to refresh the cache is to manually `mutate` it. If in any corner case scenario we don't refresh the cache, at least it gets refreshed after 3 mins.
```

Originally, it felt more natural to me to use the user profile as the default info we make available before anything else happens but from a UI loading perspective the courses list may be a good choice, too.

not sure what "profile" endpoint you're referring to. These are the currently implemented endpoints our API exposes:
<img width="700" alt="Screenshot 2023-06-29 at 12 49 36" src="https://github.com/commercetools/commercetools-docs-kit/assets/106664971/b2e7c027-4bd9-4272-9d56-948d3335cf16">

Reply:
> not sure what "profile" endpoint you're referring to. 

`POST /users/{user-id}` or a newly to be created `GET /users/{user-id}`.    Alternatively a newly to be created `POST /init-session`

Reply:
I think that calling `POST /users/{user-id}` which is an endpoint dedicated to update the user profile is even more misleading than calling the `/courses`. Let's put it like this: we need a new endpoint dedicated to session creation. Therefore I'd tag this to ticket as on hold until we have that endpoint. (I can work on that endpoint now, it should not be too hard to implement)

Reply:
> Sorry, I did not remember them. Maybe we should expose them somewhere centrally as a constant and document them once there (separately from this). It would make clear that it's a standard config.

True, I've done that in the latest commit 👍